package/MDAnalysis/core/universe.py

```diff
@@ -124,54 +126,59 @@ def __init__(self, *args, **kwargs):
             self.atoms = None
             return
 
-        self.filename = args[0]
         coordinatefile = args[1:]
 
-        topology_format = kwargs.pop('topology_format', None)
-
-        if len(args) == 1:
-            # special hacks to treat a coordinate file as a coordinate AND topology file
-            if kwargs.get('format', None) is None:
-                kwargs['format'] = topology_format
-            elif topology_format is None:
-                topology_format = kwargs.get('format', None)
-
-            # if passed a Reader, use that
-            fmt = kwargs.get('format', None)
+        # if we're given a Topology object, we don't need to parse anything
+        if isinstance(args[0], Topology):
+            self._topology = args[0]
+            self.filename = None
+        else:
+            self.filename = args[0]
+            topology_format = kwargs.pop('topology_format', None)
```

Comment:
@kain88-de I agree, and if it were completely up to me I would make it so that the `Universe` init took `topfile` followed by `trajfile`, with the possibility for `trajfile` to be a list of filenames. Because instead we use `*args` style arguments for this, it's not easy to do unless we're willing to break a very visible API piece.